rfcs/text/0408-versioning-types.md

```diff
@@ -0,0 +1,344 @@
+  "properties": {
+    "https://blockprotocol.org/@alice/types/property-type/name": {
+      "$ref": "https://blockprotocol.org/@alice/types/property-type/name/v/13"
+    },
+    "https://blockprotocol.org/@alice/types/property-type/published-on": {
+      "$ref": "https://blockprotocol.org/@alice/types/property-type/published-on/v/23"
+    },
+    "https://blockprotocol.org/@alice/types/property-type/blurb": {
+      "$ref": "https://blockprotocol.org/@alice/types/property-type/blurb/v/4"
+    }
+  },
+  "required": ["https://blockprotocol.org/@alice/types/property-type/name"]
+}
+```
+
+```json
+{
+  "https://blockprotocol.org/@alice/types/property-type/name": "The Time Machine",
+  "https://blockprotocol.org/@alice/types/property-type/published-on": "1895-05",
+  "https://blockprotocol.org/@alice/types/property-type/blurb": ...
+}
+```
+
+## Determining Type Compatibility
+
+A key part of the reasoning for picking this approach depends on the ability to determine compatibility between various types.
+
+For the purposes of this section, the following assumptions are applied:
+
+- Any given Schema X is _compatible with_ another schema Y if all possible values that satisfy Schema X also satisfy schema Y (note the directionality)
+- Any given Schema X is _equivalent to_ another Schema Y iff X is _compatible with_ Y _and_ Y is _compatible with_ X
+- All Type schemas have `additionalProperties: false` unless otherwise stated
+- 'Constraints' refer to all JSON schema keywords which affect the validation of data in the Block Protocol, these include (but are not limited to) `type`, `properties`, `minItems`, `maxItems`, `minimum`, `maximum`
+- 'Semantic Annotations' refer to all JSON schema keywords which do not affect the validation of data in the Block Protocol, these include (but are not limited to) `title`, `description`, `examples`, `default`
+- The ordering of constraints does not affect the compatibility of schemas
+
+### Data Types
+
+At the present Data Types represent completely disjoint value spaces, and the base primitive types are all **incompatible** with one another. As the [Non-Primitive Data Types RFC](https://github.com/blockprotocol/blockprotocol/pull/355) continues to be specified, it should comment on how checking compatibility will be affected.
+
+### Property Types
+
+1.  Due to how Property Types are used, they implicitly affect the structure of the data in that they require the existence of a key that matches their URI.
+1.  Property Types define their constraints through a `oneOf` field. This `oneOf` is a collection of a number of values (greater than 1) where each value is a variant of the "Property Values" (defined below).
+
+A Property Type `A` is therefore compatible with another Property Type `B` if and only if
+
+1.  They have the same base URI (which implies they are the same property type but perhaps different versions) and,
+1.  Every element of `A`'s `oneOf` is compatible with one element of `B`'s `oneOf`
```

Comment:
Nit:
```suggestion
A Property Type `A` is therefore compatible with another Property Type `B` if and only if:

1.  they have the same base URI (which implies they are the same Property Type but perhaps different versions), and
1.  every element of `A`'s `oneOf` is compatible with one element of `B`'s `oneOf`
```

Reply:
I have gone around and made the capitalisation of all of the types consistent. Thank you.

Reply:
Thanks - just want to flag it wasn't just capitalization fixes I suggested, I think the comma placement is incorrect (should be `, and` instead of `and, `). But it's just a nit pick.

Reply:
Oops missed that one, thanks for the call-out!